packages/react-app/src/hooks/useLocalState.js

```diff
@@ -0,0 +1,39 @@
+import { useCallback, useEffect, useState } from 'react';
+
+const useLocalState = (
+  initialValue,
+  key = 'omnibridge-key',
+  valueType = 'string',
+) => {
+  const storageValue = localStorage.getItem(key);
+  let castedValue = storageValue;
+
+  if (valueType === 'number') {
+    castedValue = parseInt(storageValue, 10);
+  } else if (valueType === 'boolean') {
+    castedValue = Boolean(storageValue);
+  } else if (valueType === 'object') {
+    castedValue = JSON.parse(storageValue);
+  }
+
+  const [value, setValue] = useState(castedValue || initialValue);
+
+  const updateValue = useCallback(
```

Comment:
I just noticed this now, but with every change it is automatically updated to the localStorage?
And the save callback as well as button seem redundant.

I think this would be too many calls to the localStorage, which we should avoid.
Can we change this so that only on clicking `save` button, the values will be updated to the localStorage?